async-nats/src/jetstream/kv/mod.rs

```diff
@@ -675,24 +695,14 @@ impl Store {
             })
             .await?;
 
-        let consumer_name = consumer.info.name.clone();
-
-        let mut entries = History {
+        let entries = History {
             done: consumer.info.num_pending == 0,
             subscription: consumer.messages().await?,
             prefix: self.prefix.clone(),
             bucket: self.name.clone(),
         };
 
-        let mut keys = HashSet::new();
-        while let Some(entry) = entries.try_next().await? {
-            // Filter out deleted keys
-            if !matches!(entry.operation, Operation::Purge | Operation::Delete) {
-                keys.insert(entry.key);
-            }
-        }
-        self.stream.delete_consumer(&consumer_name).await?;
```

Comment:
Please note that this is no longer required. The server was cleaning up the consumer just fine, but the bug fixed in #824 was actually causing the problem with duplicate consumers